managers/backup-manager/BackupWatcher.js

```diff
@@ -0,0 +1,151 @@
+    const plan = catalog.getPlan(changedOptions.plan_id);
+    return bm.createManager(plan)
+      .then(manager => manager.startBackup(changedOptions));
+  }
+
+  static _processAbort(changeObjectBody) {
+    const changedOptions = JSON.parse(changeObjectBody.spec.options);
+    return eventmesh.apiServerClient.getOperationOptions({
+        resourceId: changedOptions.instance_guid,
+        operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+        operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+        operationId: changedOptions.guid
+      })
+      .then(options => {
+        return Promise.try(() => {
+          const plan = catalog.getPlan(options.plan_id);
+          return bm.createManager(plan);
+        }).then(manager => manager.abortLastBackup(options));
+      });
+  }
+
+  static _processDelete(changeObjectBody) {
+    const changedOptions = JSON.parse(changeObjectBody.spec.options);
+    return eventmesh.apiServerClient.getOperationOptions({
+        resourceId: changedOptions.instance_guid,
+        operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+        operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+        operationId: changedOptions.guid
+      })
+      .then(options => {
+        return Promise.try(() => {
+          const plan = catalog.getPlan(options.plan_id);
+          return bm.createManager(plan);
+        }).then(manager => manager.deleteBackup(options));
+      });
+  }
+
+  /**
+   * @description Registers backup watcher with worker callback
+   */
+  registerWatcher() {
+    logger.info(`Registering Backup watcher`);
+    const queryString = `state in (${CONST.APISERVER.RESOURCE_STATE.IN_QUEUE},${CONST.OPERATION.ABORT},${CONST.APISERVER.RESOURCE_STATE.DELETE})`;
+    return eventmesh.apiServerClient.registerWatcher('backup', 'defaultbackup', this.worker, queryString)
+      .then(stream => {
+        logger.info(`Successfully set watcher with query string:`, queryString);
+        return setTimeout(() => {
+          try {
+            logger.info(`Refreshing stream after ${CONST.APISERVER.WATCHER_REFRESH_INTERVAL}`);
+            stream.abort();
+            return this.registerWatcher();
+          } catch (err) {
+            logger.error('Error caught in the set timout callback');
+            throw err;
+          }
+        }, CONST.APISERVER.WATCHER_REFRESH_INTERVAL);
+      })
+      .catch(e => {
+        logger.error('Failed registering watcher with error:', e);
+        throw e;
+      });
+  }
+  /**
+   * @description This method does following in order
+   * 1. Tries to acquire processing lock => In case of HA, Only one backup manager process will process the request
+   * 2. Processes the request based on the state of the changed resource
+   * 3. Release processing lock
+   * @param {object} change - Change object that comes as part of apiserver watch event
+   */
+
+  worker(change) {
+
+    logger.info('Changed Resource:', change);
```

Comment:
change log level to debug. This will lead to lots of logs.

Reply:
done